### ISSUE TITLE
question about logging

### ISSUE THREAD

**Issue by chipitsine** [haproxytech/dataplaneapi#73]
hello, I enabled logs

`--log-to=file --log-file=/var/log/dataplaneapi.log --log-level=trace --log-format=text`

I see "started" and "completed"

```
time="2020-02-13T10:00:26+03:00" level=info msg="started handling request" method=GET remote="62.168.245.226:6852" request=/login.html
time="2020-02-13T10:00:26+03:00" level=info msg="completed handling request" length=55B status=404 took="71.568µs"
```

```
time="2020-02-13T10:00:22+03:00" level=info msg="started handling request" method=PUT remote="46.17.202.14:30098" request=/v1/services/haproxy/transactions/d87c100b-1035-4ed2-8291-5d98d3a808f2
time="2020-02-13T10:00:22+03:00" level=info msg="completed handling request" length=79B status=202 took=45.294544ms
```

however, it is not clear how to detect which "completed" is related to which "started".

is it possible to add unique id ? or to combine "started" and "completed" into single log record ?

**Comment by mjuraga** [comment]
Great point, we will add this to the roadmap.

**Comment by prometherion** [comment]
Feature has been implemented on 894252182fe6dfdde5c4f2a4ceabc39a52a8de9e.

```
time="2020-12-11T12:19:48+01:00" level=info msg="started handling request" method=GET remote="127.0.0.1:39548" request=/v2/services/haproxy/configuration/global request_id=01ES8REHEZMCJT90SE7ZTKP94H
```

The `request_id` value can be assigned from the client passing the header `X-Request-Id`, otherwise an [ULID](https://github.com/oklog/ulid) one will be generated.
